scripts/getDivviIntegrators.ts

```diff
@@ -0,0 +1,168 @@
+import { writeFileSync } from 'fs'
+import yargs from 'yargs'
+import { Address, encodeEventTopics } from 'viem'
+import { LogField } from '@envio-dev/hypersync-client'
+import { paginateQuery } from './utils/hypersyncPagination'
+import { getHyperSyncClient } from './utils'
+import { NetworkId } from './types'
+import { divviRegistryAbi } from '../abis/DivviRegistry'
+import { rewardPoolAbi } from '../abis/RewardPool'
+
+const DIVVI_REGISTRY_CONTRACT_ADDRESS =
+  '0xEdb51A8C390fC84B1c2a40e0AE9C9882Fa7b7277'
+const DIVVI_INTEGRATION_REWARDS_ENTITY =
+  '0x6226ddE08402642964f9A6de844ea3116F0dFc7e'
+
+async function getArgs() {
+  const argv = await yargs
+    .env('')
+    .option('output-file', {
+      alias: 'o',
+      description: 'output file',
+      type: 'string',
+    })
+    .option('reward-pool', {
+      description: 'RewardPool contract address',
+      type: 'string',
+      demandOption: true,
+    })
+    .option('reward-amount', {
+      description: 'reward amount for integration',
+      type: 'string',
+    }).argv
+
+  return {
+    output: argv['output-file'] ?? 'divvi-integrator-rewards.csv',
+    rewardPoolContractAddress: argv['reward-pool'],
+    rewardAmount: argv['reward-amount'],
+  }
+}
+
+function removeDuplicates<T>(arr: T[]): T[] {
+  const seen = new Set<T>()
+  return arr.filter((item) => {
+    if (seen.has(item)) {
+      return false
+    }
+    seen.add(item)
+    return true
+  })
+}
+
+async function getDivviIntegrators({
+  rewardPoolContractAddress,
+}: {
+  rewardPoolContractAddress: string
+}): Promise<Address[]> {
+  const usersThatHaveIntegrated: Address[] = []
+  const usersThatHaveReceivedRewards = new Set<Address>()
+  const usersThatHaveRegisteredAgreements = new Set<Address>()
```

Comment:
can we change `user` -> `consumer` to be consistent with the terminology (since `user` should refer to the end user, this was a bit confusing to read)